src/openforms/forms/admin/views.py

```diff
@@ -0,0 +1,76 @@
+from datetime import date
+
+from django import forms
+from django.contrib.messages.views import SuccessMessageMixin
+from django.contrib.postgres.forms import SimpleArrayField
+from django.core.exceptions import PermissionDenied
+from django.http import FileResponse
+from django.shortcuts import get_object_or_404
+from django.urls import reverse_lazy
+from django.utils.translation import ugettext_lazy as _
+from django.views.generic import DetailView
+from django.views.generic.edit import FormView
+
+from ..models.form import Form, FormsExport
+from .tasks import process_forms_export
+from .tokens import exported_forms_token_generator
+
+
+class FormsUUIDsForm(forms.Form):
+    forms_uuids = SimpleArrayField(forms.UUIDField(), required=True)
+    username = forms.CharField(max_length=150, required=True)
+
+
+class ExportFormsForm(forms.Form):
+    forms_uuids = SimpleArrayField(forms.UUIDField(), widget=forms.HiddenInput)
+    username = forms.CharField(max_length=150, widget=forms.HiddenInput)
+    email = forms.EmailField()
+
+    def clean_forms_uuids(self):
+        if not Form.objects.filter(uuid__in=self.cleaned_data["forms_uuids"]).exists():
+            raise forms.ValidationError(_("Invalid form uuids."), code="invalid")
+        return self.cleaned_data["forms_uuids"]
+
+
+class ExportFormsView(SuccessMessageMixin, FormView):
+    template_name = "admin/forms/form/export.html"
+    form_class = ExportFormsForm
+    success_url = reverse_lazy("admin:forms_form_changelist")
+    success_message = _("Success! You will receive an email when your export is ready.")
+
+    def get_initial(self):
+        initial_data = super().get_initial()
+        form = FormsUUIDsForm(
+            data={
+                "forms_uuids": self.request.session.get("forms_uuids"),
+                "username": self.request.user.username,
+            }
+        )
+        if form.is_valid():
+            initial_data.update(form.cleaned_data)
+        return initial_data
```

Comment:
This is a weird construction, but if you refactor the action/view function this will not be relevant anymore.

However, I'd expect that the initial data would just be `"forms_uuids": self.request.session.get("forms_uuids")` and done. Not sure why things are still wrapped in another form that is not involved in the view itself.

The `username` construct is also odd - I'd expect to just pass around `request.user` or for celery tasks, `request.user.id`